pkg/filter/proxy/masterslavereader.go

```diff
@@ -66,7 +68,9 @@ func (mr *masterReader) Read(p []byte) (n int, err error) {
 	}
 
 	if err == io.EOF {
-		close(mr.buffChan)
+		mr.closeChanOnce.Do(func() {
+			close(mr.buffChan)
+		})
```

Comment:
If non-EOF error happens, do we need to take care of `mr.buffChan`?

Reply:
No